src/nomad_simulations/outputs.py

```diff
@@ -15,38 +15,153 @@
+from .model_system import ModelSystem
+from .numerical_settings import SelfConsistency
 
 
-class Outputs(ArchiveSection):
-    """ """
+class BaseOutputs(ArchiveSection):
+    """
+    Base section to define the output properties of a simulation. This is used as a placeholder for both
+    final `Outputs` properties and the self-consistent (SCF) steps, see `Outputs` base section definition.
+    """
+
+    # TODO add time quantities
 
     normalizer_level = 2
 
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the output property. This is used for easier identification of the property and is conneceted
+        with the class name of each output property class, e.g., `'ElectronicBandGap'`, `'ElectronicBandStructure'`, etc.
+        """,
+        a_eln=ELNAnnotation(component='StringEditQuantity'),
+    )
+
+    orbitals_state_ref = Quantity(
+        type=OrbitalsState,
+        description="""
+        Reference to the `OrbitalsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    atoms_state_ref = Quantity(
+        type=AtomsState,
+        description="""
+        Reference to the `AtomsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    model_system_ref = Quantity(
+        type=ModelSystem,
+        description="""
+        Reference to the `ModelSystem` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the output property is derived from other output properties. We make
+        the distinction between directly parsed, derived, and post-processing output properties:
+            - Directly parsed: the output property is directly parsed from the simulation output files.
+            - Derived: the output property is derived from other output properties. No extra numerical settings
+                are required to calculate the output property.
+            - Post-processing: the output property is derived from other output properties. Extra numerical settings
+                are required to calculate the output property.
+        """,
+    )
+
+    outputs_ref = Quantity(
+        type=Reference(SectionProxy('BaseOutputs')),
+        description="""
+        Reference to the `BaseOutputs` section from which the output property was derived. This is only
+        relevant if `is_derived` is set to True.
+        """,
+    )
+
+    def check_is_derived(self, is_derived: bool, outputs_ref) -> Optional[bool]:
+        if not is_derived:
+            if outputs_ref is not None:
+                return True
+            return False
+        elif is_derived and outputs_ref is not None:
+            return True
+        return None
+
```

Comment:
Not sure if I follow the logic here (maybe also add a small pydocs):

This function is used to set `is_derived`, correct?
If `is_derived` is already set (i.e. `is not None`), only then it should trigger.

Moreover, if `outputs_ref` fully covers `is_derived`, then you don't need the latter.

Reply:
One more point, I'm going to ask Area D for "write-protected" quantities, i.e. if the value has been set, it can't be updated. This would help make the normalization leaner.

Reply:
No, because it might be missing `outputs_ref`, hence the `None` return describes the "missing refs" case. I added a better description after you reviewed it, sorry.

Reply:
For number 3., maybe some wrong setting of `is_derived` or some missing `outputs_ref` when normalizing? In this case, `return None` is happening